xen/drivers/passthrough/arm/io-pgtable-arm.c

```diff
@@ -195,11 +238,21 @@ struct arm_lpae_io_pgtable {
 	unsigned long		pg_shift;
 	unsigned long		bits_per_level;
 
-	void			*pgd;
+	/* Xen: We deal with domain pages. */
+	struct page_info	*pgd;
 };
 
 typedef u64 arm_lpae_iopte;
 
+/*
+ * Xen: Overwrite Linux functions that are in charge of memory
+ * allocation/deallocation by Xen ones. The main reason is that we want to
+ * operate with domain pages and as the result we have to use Xen's API for this.
+ * Taking into account that Xen's API deals with struct page_info *page
+ * modify all depended code. Also keep in mind that the domain pages must be
+ * mapped just before using it and unmapped right after we completed.
+ */
+#if 0
```

Comment:
No. Nothing need to be changed for now. There are two possible ways:
1. All code under #if 0 should be dropped.
2. The "#if 0" should be left as is like SMMU does.